services/src/handlers/workflows.rs

```diff
@@ -240,19 +257,299 @@ async fn get_workflow_provenance<C: Context>(
+        dataset,
+        upload,
+    }))
+}
+
+async fn build_geotiff<C: Context>(
+    processor: TypedRasterQueryProcessor,
+    file_path: &Path,
+    query_rect: RasterQueryRectangle,
+    query_ctx: <C as Context>::QueryContext,
+    no_data_value: Option<f64>,
+    request_spatial_ref: SpatialReference,
+    tile_limit: Option<usize>,
+) -> Result<()> {
+    match processor {
+        geoengine_operators::engine::TypedRasterQueryProcessor::U8(p) => {
+            raster_stream_to_geotiff(
+                file_path,
+                p,
+                query_rect,
+                query_ctx,
+                no_data_value,
+                request_spatial_ref,
+                tile_limit,
+            )
+            .await
+        }
+        geoengine_operators::engine::TypedRasterQueryProcessor::U16(p) => {
+            raster_stream_to_geotiff(
+                file_path,
+                p,
+                query_rect,
+                query_ctx,
+                no_data_value,
+                request_spatial_ref,
+                tile_limit,
+            )
+            .await
+        }
+        geoengine_operators::engine::TypedRasterQueryProcessor::U32(p) => {
+            raster_stream_to_geotiff(
+                file_path,
+                p,
+                query_rect,
+                query_ctx,
+                no_data_value,
+                request_spatial_ref,
+                tile_limit,
+            )
+            .await
+        }
+        geoengine_operators::engine::TypedRasterQueryProcessor::I16(p) => {
+            raster_stream_to_geotiff(
+                file_path,
+                p,
+                query_rect,
+                query_ctx,
+                no_data_value,
+                request_spatial_ref,
+                tile_limit,
+            )
+            .await
+        }
+        geoengine_operators::engine::TypedRasterQueryProcessor::I32(p) => {
+            raster_stream_to_geotiff(
+                file_path,
+                p,
+                query_rect,
+                query_ctx,
+                no_data_value,
+                request_spatial_ref,
+                tile_limit,
+            )
+            .await
+        }
+        geoengine_operators::engine::TypedRasterQueryProcessor::F32(p) => {
+            raster_stream_to_geotiff(
+                file_path,
+                p,
+                query_rect,
+                query_ctx,
+                no_data_value,
+                request_spatial_ref,
+                tile_limit,
+            )
+            .await
+        }
+        geoengine_operators::engine::TypedRasterQueryProcessor::F64(p) => {
+            raster_stream_to_geotiff(
+                file_path,
+                p,
+                query_rect,
+                query_ctx,
+                no_data_value,
+                request_spatial_ref,
+                tile_limit,
+            )
+            .await
+        }
```

Comment:
don't we have a macro for that?

Reply:
unfortunately not. Our macro requires the function to work on ALL RasterDataTypes but `raster_stream_to_geotiff` doesn't work on I8 because Gdal doesn't know signed bytes

Reply:
cf7630eb8e3684f9b0f069b40035c3fa26971b8e